shell/components/form/ArrayList.vue

```diff
@@ -266,6 +274,17 @@ export default {
                 @paste="onPaste(idx, $event)"
                 @input="queueUpdate"
               />
+              <LabeledInput
```

Comment:
This only makes rules work for a small subset of the ArrayList functionality. I think rules need to be evaluated within arraylist itself otherwise it won't work with slots, multiline or regular input.

Reply:
With slots, we'll be able to pass in rules directly to the input component where they're passed into the slot and the arraylist itself won't even need the rules passed into it. As far as multi-line input, it doesn't look like we actually use this option for arraylist anywhere in the codebase (I considered ripping it out actually but figured that'd be stepping outside of the scope of the ticket a bit much). For regular input, it should never actually be used if there are more than 0 rules passed in since the rules array length is what I'm actually evaluating to know if I should be using labeled-input or the standard input.

I _could_ refactor this component to add in the labeled tool-tip to the basic html input on arraylist that we're using here but that'd basically just be copy/pasta from labeled-input which already knows how to handle rules and really just needs the compact option turned off in order to fit into this form correctly. I could also just rip out the basic html input altogether but I'd run the risk of messing up any deep targeted CSS that might be using arraylist and potentially increase the QA overhead (I didn't do a full analysis to figure out how common that is though); instead I chose to basically gate it behind the presence of rules so that I know it's only used in one place for now.

Reply:
I think you should file an issue to do the refactor so we can remove the dead code. Once the issue is filed so we can track it I'm fine to ship it.

Reply:
https://github.com/rancher/dashboard/issues/6507